src/StaticArraysCore.jl

```diff
@@ -44,21 +44,28 @@ const StaticVecOrMat{T} = Union{StaticVector{<:Any, T}, StaticMatrix{<:Any, <:An
 # The ::Tuple variants exist to make sure that anything that calls with a tuple
 # instead of a Tuple gets through to the constructor, so the user gets a nice
 # error message
-Base.@pure tuple_length(T::Type{<:Tuple}) = length(T.parameters)
-Base.@pure tuple_length(T::Tuple) = length(T)
-Base.@pure tuple_prod(T::Type{<:Tuple}) = length(T.parameters) == 0 ? 1 : *(T.parameters...)
-Base.@pure tuple_prod(T::Tuple) = prod(T)
-Base.@pure tuple_minimum(T::Type{<:Tuple}) = length(T.parameters) == 0 ? 0 : minimum(tuple(T.parameters...))
-Base.@pure tuple_minimum(T::Tuple) = minimum(T)
+tuple_length(T::Tuple) = length(T)
+tuple_prod(T::Tuple) = prod(T)
+tuple_minimum(T::Tuple) = minimum(T)
+
+tuple_svec(::Type{T}) where {T<:Tuple} = T.parameters
+
+# Julia `Base` provides a function for this purpose since v1.1: `fieldtypes`.
+# We don't use it yet, though, because it wrecks type inference with Julia
+# v1.6.
+Base.@pure tuple_tuple(::Type{T}) where {T<:Tuple} = (tuple_svec(T)...,)
```

Comment:
FTR: I suppose that another possible implementation would go via `fieldtype` (no `s`!) and `ntuple`. But I guess this is OK for now.

Reply:
It seems OK. We could also have a `@pure` implementation for Julia 1.6 and a `fieldtypes` implementation for later version.